source/Halibut.Tests/Diagnostics/ExceptionReturnedByHalibutProxyExtensionMethodFixture.cs

```diff
@@ -0,0 +1,211 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Halibut.Diagnostics;
+using Halibut.ServiceModel;
+using Halibut.Tests.TestServices;
+using Halibut.Tests.Util;
+using Halibut.Transport.Proxy;
+using NUnit.Framework;
+
+namespace Halibut.Tests.Diagnostics
+{
+    public static class ExceptionReturnedByHalibutProxyExtensionMethodFixture
```

Comment:
I don't think it is possible to just throw a "ThisWasANetworkError" in a consistent way, so I have added lots of tests to try to verify the extension method actually works.